resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsPollingSnapshotSource.scala

```diff
@@ -55,14 +56,17 @@ private[spark] class ExecutorPodsPollingSnapshotSource(
   private class PollRunnable(applicationId: String) extends Runnable {
     override def run(): Unit = Utils.tryLogNonFatalError {
       logDebug(s"Resynchronizing full executor pod state from Kubernetes.")
-      snapshotsStore.replaceSnapshot(kubernetesClient
+      val pods = kubernetesClient
         .pods()
         .withLabel(SPARK_APP_ID_LABEL, applicationId)
         .withLabel(SPARK_ROLE_LABEL, SPARK_POD_EXECUTOR_ROLE)
         .withoutLabel(SPARK_EXECUTOR_INACTIVE_LABEL, "true")
-        .list()
-        .getItems
-        .asScala.toSeq)
+      val list = if (conf.get(KUBERNETES_EXECUTOR_API_POLLING_WITH_RESOURCE_VERSION)) {
+        pods.list(new ListOptionsBuilder().withResourceVersion("0").build())
+      } else {
+        pods.list()
+      }
+      snapshotsStore.replaceSnapshot(list.getItems.asScala.toSeq)
```

Comment:
Would it make sense to check the snapShot resource verison returned and only replace if it's newer than the previous snapShot? Since in HA fail-over we could go back in time which might be confusing to Spark.

Reply:
Thanks. Yes, definitely, you are right. That's the reason why I added this as `false` with a warning in the conf. It depends on the K8s API server implementation and the cluster situations.

Reply:
> Would it make sense to check the snapShot resource verison returned and only replace if it's newer than the previous snapShot? 

+1 to doing this so that setting this to `true` will still not affect Spark if a given k8s-api-server is catching up to latest state from etcd.

Reply:
BTW, I'm thinking about that as a next step because the same resource versions already happen in the AS-IS status. `pods.list()` just returns the latest one. It doesn't return a new value always.

Reply:
I'm taking back my words. I'll revise this PR, ASAP.

Reply:
Thank you @dongjoon-hyun